vault/resource_identity_entity_alias_test.go

```diff
@@ -139,3 +164,31 @@ resource "vault_identity_entity_alias" "entity-alias-dupe" {
 
 	return ret
 }
+
+func testAccIdentityEntityAliasMetadataConfig(entityName string) string {
+	entityId := "A"
+
+	ret := fmt.Sprintf(`
+resource "vault_identity_entity" "entityA" {
+  name = "%s-A"
+  policies = ["test"]
+  metadata = {
+    version = "1"
+  }
+}
+
+resource "vault_auth_backend" "githubA" {
+  type = "github"
+  path = "githubA-%s"
+}
+
+resource "vault_identity_entity_alias" "entity-alias" {
+  name = vault_identity_entity.entity%s.name
+  mount_accessor = vault_auth_backend.github%s.accessor
+  canonical_id = vault_identity_entity.entity%s.id
+  custom_metadata = vault_identity_entity.entity%s.metadata
```

Comment:
Do we have tests without the `custom_metadata` being set?

Reply:
Added another step in `TestAccIdentityEntityAlias_Metadata` that removes metadata after it was set in the previous step